webapp/src/errorList.tsx

```diff
@@ -29,27 +29,37 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
     }
 
     render() {
+        const showAvailableErrors = this.state.errors.length != 0;
         const showCollapseButton = true;
         const collapseTooltip = lf("Collapse Error List");
         function errorKey(error: pxtc.KsDiagnostic): string {
             // React likes have a "key" for each element so that it can smartly only
             // re-render what changes. Think of it like a hashcode/
             return `${error.messageText}-${error.fileName}-${error.line}-${error.column}`
         }
-        return <div className="errorList" >
-            {showCollapseButton &&
+            
+        return <div className="errorList" > 
+            {showCollapseButton && showAvailableErrors &&
                 <sui.Button id='toggleErrorList' className={`toggleErrorList collapse-button large`}
-                    icon={`inverted chevron ${this.state.isCollapsed ? 'up' : 'down'}`}
-                    title={collapseTooltip} onClick={this.onCollapseClick} />}
-            <div className="errorListInner" hidden={this.state.isCollapsed}>
-                {
+                icon={`inverted chevron ${this.state.isCollapsed ? 'up' : 'down'}`}
+                title={collapseTooltip} onClick={this.onCollapseClick} />}
+            {showAvailableErrors
+                ? <div className="errorListInner" hidden={this.state.isCollapsed}>
+                    {
                     (this.state.errors || []).map(e =>
                         <div key={errorKey(e)}>{`${e.messageText} - [line ${e.line + 1}: col ${e.column}]`}</div>)
-                }
-            </div>
+                    }
+                  </div>
+                : <div className="errorListInner" id="noErrorsMessage">{lf("You have no errors")}</div>
```

Comment:
typically instead of adding an id to a div dynamically you would just add a class. The "className" can be a list of space seperated names

Reply:
Usually we think of an id as a constant. A HTML element either has one or it doesn't but it doesn't usually change.